src/measure_theory/function/convergence_in_measure.lean

```diff
@@ -0,0 +1,272 @@
+        (inv_le_one one_le_two) _,
+      rw [sub_le_iff_le_add, ← nat.cast_add_one, nat.cast_le],
+      exact (le_tsub_add.trans (add_le_add_right (le_max_right _ _) 1)).trans
+        (add_le_add_right hn_ge 1) },
+    refine le_trans _ h_inv_n_le_k,
+    rw [set.mem_compl_iff, set.nmem_set_of_eq, not_le] at hNx,
+    exact hNx.le },
+  rw ae_iff,
+  refine measure_mono_null (λ x, _) hμs,
+  rw [set.mem_set_of_eq, ← @not_not (x ∈ s), not_imp_not],
+  exact h_tendsto x,
+end
+
+end
+
+section
+
+variables [measurable_space E] [normed_group E] [borel_space E] [has_measurable_sub₂ E] {p : ℝ≥0∞}
+variables {f : ℕ → α → E} {g : α → E}
+
+/-- Auxiliary lemma for `tendsto_in_measure_of_tendsto_snorm`. -/
+lemma tendsto_in_measure_of_tendsto_snorm_of_measurable
+  (hp_ne_zero : p ≠ 0) (hp_ne_top : p ≠ ∞)
+  (hf : ∀ n, measurable (f n)) (hg : measurable g)
+  (hfg : tendsto (λ n, snorm (f n - g) p μ) at_top (𝓝 0)) :
+  tendsto_in_measure μ f g :=
+begin
+  intros ε hε,
+  replace hfg := ennreal.tendsto.const_mul (tendsto.ennrpow_const p.to_real hfg)
+    (or.inr $ @ennreal.of_real_ne_top (1 / ε ^ (p.to_real))),
+  simp only [mul_zero, ennreal.zero_rpow_of_pos (ennreal.to_real_pos hp_ne_zero hp_ne_top)] at hfg,
+  rw ennreal.tendsto_at_top_zero at hfg ⊢,
+  intros δ hδ,
+  obtain ⟨N, hN⟩ := hfg δ hδ,
+  refine ⟨N, λ n hn, le_trans _ (hN n hn)⟩,
+  rw [ennreal.of_real_div_of_pos (real.rpow_pos_of_pos hε _), ennreal.of_real_one, mul_comm,
+    mul_one_div, ennreal.le_div_iff_mul_le _ (or.inl (ennreal.of_real_ne_top)), mul_comm],
+  { convert mul_meas_ge_le_pow_snorm' μ hp_ne_zero hp_ne_top ((hf n).sub hg) (ennreal.of_real ε),
+    { exact (ennreal.of_real_rpow_of_pos hε).symm },
+    { ext x,
+      rw [dist_eq_norm, ← ennreal.of_real_le_of_real_iff (norm_nonneg _),
+          of_real_norm_eq_coe_nnnorm] } },
+  { rw [ne, ennreal.of_real_eq_zero, not_le],
+    exact or.inl (real.rpow_pos_of_pos hε _) },
+end
+
+/-- Convergence in Lp implies convergence in measure. -/
+lemma tendsto_in_measure_of_tendsto_snorm
+  (hp_ne_zero : p ≠ 0) (hp_ne_top : p ≠ ∞)
+  (hf : ∀ n, ae_measurable (f n) μ) (hg : ae_measurable g μ)
+  (hfg : tendsto (λ n, snorm (f n - g) p μ) at_top (𝓝 0)) :
+  tendsto_in_measure μ f g :=
+begin
+  refine tendsto_in_measure.congr (λ i, (hf i).ae_eq_mk.symm) hg.ae_eq_mk.symm _,
+  refine tendsto_in_measure_of_tendsto_snorm_of_measurable hp_ne_zero hp_ne_top
+    (λ i, (hf i).measurable_mk) hg.measurable_mk _,
+  have hf_eq_ae : ∀ᵐ x ∂μ, ∀ n, (hf n).mk (f n) x = f n x,
+    from ae_all_iff.mpr (λ n, (hf n).ae_eq_mk.symm),
+  have : (λ n, snorm ((hf n).mk (f n) - ae_measurable.mk g hg) p μ) = (λ n, snorm (f n - g) p μ),
+  { ext1 n, refine snorm_congr_ae (eventually_eq.sub (hf n).ae_eq_mk.symm hg.ae_eq_mk.symm), },
+  rw this,
+  exact hfg,
+end
+
+/-- Convergence in Lp implies convergence in measure. -/
+lemma tendsto_in_measure_of_tendsto_Lp [second_countable_topology E] [hp : fact (1 ≤ p)]
```

Comment:
could you also write a version of this lemma assuming that the measure is finite, and allowing p = infinity?